YR/Add-Bearer-Token-Authentication/Jamfv2/CIAC-10058 (#33646)
demisto/content: Packs/jamf/Integrations/jamfV2/jamfV2_description.md

This integration supports three authentication methods:

> 1. **Basic authentication:** Deprecated starting from version *10.35.0* of Jamf. For more information, click [here](https://developer.jamf.com/jamf-pro/docs/classic-api-authentication-changes).

Reviewer: ```suggestion
1. **Basic authentication:** Deprecated starting from version *10.35.0* of Jamf. For more information, click [here](https://developer.jamf.com/jamf-pro/docs/classic-api-authentication-changes).
Note: This method is used as fallback if other authentication method failed, which will result in general failure if basic authentication is disabled in JAMF.
```

Author: This is not exactly correct, since it will be used as a fallback only for the second method, not "if other authentication method failed", and I alredy wrote that at the bottem.

2. **Bearer token based on basic authentication:** 
Use `username and password`.

> The username and password of a user with relevant permissions to your Jamf Pro admin console

Reviewer 2: ```suggestion
The username and password of a user with relevant permissions to your Jamf Pro admin console.
```

3. **Bearer token based on client credentials:**
Use `client ID and client secret`.

> See this [link](https://learn.jamf.com/en-US/bundle/jamf-pro-documentation-current/page/API_Roles_and_Clients.html) how te retrieve a client id and client secret.

Reviewer 2: ```suggestion
See this [link](https://learn.jamf.com/en-US/bundle/jamf-pro-documentation-current/page/API_Roles_and_Clients.html) for how to retrieve a client ID and client secret.
```

> *Important*: Before creating an AI client, ensure that you create an API role with all the privileges as mentioned in the integration documentation. Then, associate the API client with that role.

Reviewer 2: ```suggestion
*Important*: Before creating an API client, ensure that you create an API role with all the privileges as mentioned in the integration documentation. Then, associate the API client with that role.
```

> When using the first two methods, provide a username and password. The system will automatically attempt to generate a token (second method), and if unsuccessful, it will fallback to using the first method.

Reviewer 2: ```suggestion
When using the first two methods, provide a username and password. The system will automatically attempt to generate a token using he Bearer token based on basic authentication method. If unsuccessful, it will fallback to using the Basic authentication method.
```

> When using the first two methods, provide a username and password. The system will automatically attempt to generate a token (second method), and if unsuccessful, it will fallback to using the first method, but only if basic authentication is permitted by the user.

Reviewer: ```suggestion
When using the first two methods, a username and password must be provided.
```

Author: Done
